### DISCUSSION TITLE
Is next auth a good fit for seperate custom backend issuing tokens?

### DISCUSSION THREAD

**Discussion by lindeberg** [nextauthjs/next-auth#3030] [Help] [answered]
I've got an ASP.NET Core Web API with endpoints for issuing and refreshing tokens. I'm trying to figure out if next auth is a good fit for this. This is what I've got so far. Not quite working yet.

![image](https://user-images.githubusercontent.com/15940340/138840453-7dcedf77-3cab-4a26-8570-7d761c4caa4e.png)
![image](https://user-images.githubusercontent.com/15940340/138840424-4f980a23-638f-4ecb-b023-4fec76a6568a.png)

**Comment by balazsorban44** [comment, marked as answer]
hi, this is how any of the OAuth providers work, so yes. you might want to look into a framework like IdentityServer https://duendesoftware.com/ that gives YOU an openid connect compliant Identity Provider, rather than building your own solution, especially if you don't have the resources/expertise in auth

**Reply by lindeberg** [reply]
Cool, thanks!

I want to be able to just have a log in component with email address and password within my app without getting redirected anywhere. I can see big ones like AirBnb/Facebook doing that and just seems like a better user experience. Now all these OpenId Connect solutions wants me to redirect somewhere. Is there some security benefit? Why then are the big ones not doing it?

**Reply by balazsorban44** [reply]
there is absolutely a benefit. whenever you start storing passwords yourself, you will introduce extra responsibility related to security.

big oauth providers have entire teams making sure that the password is managed correctly, also adding a lot of  protection against DDoS and probably many other type of attacks I might not even be aware of.

without proper knowledge, we advocate against running your own solution.
